go/mysql/flavor_mariadb.go

```diff
@@ -152,6 +153,36 @@ func (mariadbFlavor) status(c *Conn) (SlaveStatus, error) {
 	return parseMariadbSlaveStatus(resultMap)
 }
 
+// masterStatus is part of the Flavor interface.
+func (m mariadbFlavor) masterStatus(c *Conn) (MasterStatus, error) {
+	qr, err := c.ExecuteFetch("SHOW MASTER STATUS", 100, true /* wantfields */)
+	if err != nil {
+		return MasterStatus{}, err
+	}
+	if len(qr.Rows) == 0 {
+		// The query returned no data, meaning the server
+		// is not configured as a master.
+		return MasterStatus{}, ErrNotMaster
+	}
+
+	resultMap, err := resultToMap(qr)
+	if err != nil {
+		return MasterStatus{}, err
+	}
+
+	// SHOW MASTER STATUS does not return the executed GTIDSet for MariaDB, so we patch it in.
+	qr, err = c.ExecuteFetch("SELECT @@GLOBAL.gtid_binlog_pos", 1, false)
```

Comment:
Could we just call `m.masterGTIDSet(c)`?

Reply:
I had thought of doing this but the downside here is that it already gets turned into a `Position` for us. I wanted to encapsulate that responsibility into the flavor specific parsing function.